### PR TITLE
Fix CupertinoDatePicker width on wide screens

### LINKED ISSUE #54241: CupertinoDatePicker is curving the wrong way when screen is wide
On head v1.18.1-pre.2

Use a basic CupertinoDatePicker
https://github.com/xster/flutter-test/tree/master/t156_picker

The horizontal curvature seems to be width dependent. 

Screen Shot 2020-04-07 at 4 16 22 PM

When it's too wide, it starts to curve backwards

Screen Shot 2020-04-07 at 4 16 52 PM

### PR DESCRIPTION
**Pull request by TahaTesser**
This fixes https://github.com/flutter/flutter/issues/54241

Instead of providing all the width available, here we're limiting width after a certain size and it is not needed (this makes the wheels to be aligned properly instead of curving too much on a wide screen).

#### Before

Screen Shot 2021-11-15 at 01 16 01

#### After

Screen Shot 2021-11-15 at 00 34 10

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/93629

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/93629

**Comment by skia-gold** [comment]
Gold has detected about 4 new digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/93629

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 4.
View them at https://flutter-gold.skia.org/cl/github/93629

**Comment by TahaTesser** [comment]
@HansMuller
This is ready to be reviewed, I tried my best to come up with the cleanest solution tho this still has a tiny gold test difference.
Let me know if you think this is the right way or some new direction can be taken.

File: packages/flutter/lib/src/cupertino/date_picker.dart [hunk 1]
```diff
@@ -1043,12 +1043,24 @@ class _CupertinoDatePickerDateTimeState extends State<CupertinoDatePicker> {
...
-            textDirectionFactor: textDirectionFactor,
-          ),
-          children: pickers,
+        child: LayoutBuilder(
+          builder: (BuildContext context, BoxConstraints constraints) {
+            return SizedBox(
+              width: constraints.maxWidth,
+              child: Center(
+                child: ConstrainedBox(
+                  constraints: const BoxConstraints(maxWidth: 500.0),
```
Comment (HansMuller) [inline packages/flutter/lib/src/cupertino/date_picker.dart:1052]: This seems like a dangerous thing to do. What if the device's textScaleFactor is increased or the default font is changed?

**Comment by TahaTesser** [comment]
Closing this for now, to be fixed when i study the core framework deeply

**Pull request closed without merging**